utils/downloads.go

```diff
@@ -0,0 +1,76 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/cheggaaa/pb.v1"
+)
+
+var snapshots = map[string]string{
+	"mainnet": "https://juno-snapshot.s3.us-east-2.amazonaws.com/mainnet/juno_mainnet_v0.5.0_136902.tar",
+	"goerli":  "https://juno-snapshot.s3.us-east-2.amazonaws.com/goerli/juno_goerli_v0.5.0_839969.tar",
+	"goerli2": "https://juno-snapshot.s3.us-east-2.amazonaws.com/goerli2/juno_goerli2_v0.5.0_135973.tar",
+}
+
+const (
+	downloadBufferSize = 8192 // 8KB
+	dirPermissionMod   = 0o770
+)
+
+func DownloadFile(ctx context.Context, network, location string) error {
```

Comment:
Is there a way we can test this? Maybe using `net/http/httptest`? Would be really nice to have tests for such a large function.

Reply:
I thought about that, but I'm not sure if we need to unit test a download file ?

Reply:
Better to unit test it imo. Better than coming back and debugging it later. Also ensures we don't break things if we modify the function in the future.

Reply:
Thinking out loud: I feel like this is a pretty common operation. Maybe there's a widely used and well-tested package that does something similar? I feel like that would be ideal.

Reply:
I did some research about that and I found [this](https://github.com/cavaliergopher/grab). Tho, I thought the download process is a straightforward task that can be accomplished using the standard library(`http.Get()` && `io.Copy` && `out.Write`).

Reply:
For the unit test, I agree, I will add a unit test for it.

Reply:
That's fair. Let's go with that.